xls2xml/xls2xml/XLSReader.py

```diff
@@ -0,0 +1,153 @@
+                continue
+
+            # Check number of rows
+            worksheet = self.workbook[title]
+            if worksheet.max_row < 2:
+                continue
+
+            # Check required headers are present
+            self.headers[title] = [cell.value if cell.value is None else cell.value.strip() for cell in worksheet[1]]
+            required_headers = self.xls_conf[title].get(REQUIRED_HEADERS_KEY_NAME, [])
+            if set(required_headers) <= set(self.headers[title]): # issubset
+                self.worksheets.append(title)
+
+        return self.worksheets
+
+    def get_headers_by_worksheet(self, worksheet):
+        """
+        :param worksheet: the name of the worksheet
+        :type worksheet: basestring
+        :return: the list of valid worksheet names in the Excel file
+        :rtype: list
+        """
+        worksheets = self.valid_worksheets()
+        if worksheet not in worksheets:
+            print 'Worksheet '+worksheet+' is not available or not valid!'
+            return []
+
+        return [x for x in self.headers[worksheet] if x is not None]
+
+    def next(self):
+        """
+        Retrieve next data row
+
+        :param worksheet: the name of the worksheet
+        :type worksheet: basestring
+        :return: A hash containing all the REQUIRED and OPTIONAL fields as keys
+                and the corresponding data as values
+        :rtype: dict
+        """
+        if self.worksheets is None:
+            self.valid_worksheets()
+
+        worksheet = self.active
+        if worksheet is None:
+            print 'No worksheet is specified!'
+            raise StopIteration
+
+        if worksheet not in self.worksheets:
+            print 'Worksheet ' + worksheet + ' is not valid!'
+            raise StopIteration
+
+        if worksheet not in self.row_offset:
+            self.row_offset[worksheet] = 1
+        self.row_offset[worksheet] += 1
+
+        work_sheet = self.workbook[worksheet]
+        required_headers = self.xls_conf[worksheet].get(REQUIRED_HEADERS_KEY_NAME, [])
+        optional_headers = self.xls_conf[worksheet].get(OPTIONAL_HEADERS_KEY_NAME, [])
+
+        for row in work_sheet.iter_rows(min_row=self.row_offset[worksheet]):
+            num_cells = 0
+            for cell in row:
+                num_cells += 1
+
+            data = {}
+            has_notnull = False
+            for header in required_headers+optional_headers:
+                header_index = num_cells
+                if header in self.headers[worksheet]:
+                    header_index = self.headers[worksheet].index(header)
+                if header_index >= num_cells:
+                    data[header] = None
+                    continue
+
+                cell = row[header_index]
+                if cell.value is not None:
+                    has_notnull = True
+
+                if isinstance(cell.value, unicode):
+                    data[header] = cell.value.encode('ascii', 'ignore')
```

Comment:
Why? It won't be possible to encode every Unicode character correctly.

Reply:
This is no longer needed I should have removed it earlier. Changed in ceccb29